### ISSUE TITLE
How to use HydratedBloc.storage in blocTest?

### ISSUE THREAD

**Issue by data987** [felangel/bloc#2022]
**Describe the bug**
When I run the tests without HydratedBloc the console throws me this message:

![Screen Shot 2020-12-17 at 6 22 00 PM](https://user-images.githubusercontent.com/8518699/102555431-c33fae00-4094-11eb-93e9-97a753e867f7.png)

And when I Initialize HydratedBloc.storage:

![Screen Shot 2020-12-17 at 6 22 40 PM](https://user-images.githubusercontent.com/8518699/102555473-dc485f00-4094-11eb-9933-84fcdb94f9ea.png)
 
Then run the tests I received this:

![Screen Shot 2020-12-17 at 6 23 37 PM](https://user-images.githubusercontent.com/8518699/102555539-fe41e180-4094-11eb-9b28-37c59e3e911e.png)

I'm missing something?

**Comment by felangel** [comment]
Hi @data987 👋 
Thanks for opening an issue!

You should provide a mock storage in your test rather than using the real one. Check out the hydrated bloc initialization helper at https://github.com/felangel/bloc/blob/master/examples/flutter_weather/test/helpers/hydrated_bloc.dart which is used in the weather example tests at https://github.com/felangel/bloc/blob/17b922b9bda38a06550ff648a1a4bed51ba7b982/examples/flutter_weather/test/theme/cubit/theme_cubit_test.dart#L21.

Hope that helps 👍

**Comment by data987** [comment]
Perfect! that works! thanks for your help

**Comment by sabifa** [comment]
I stumbled upon this issue today because I wanted to test a HydratedBloc. Unfortunately @felangel's linked code uses the setter HydratedBloc.storage, which does not seem to be available in the latest version (8.1.0).
```dart
HydratedBloc.storage = hydratedStorage;
```

I am a bit confused because it seems like the setter is only available in the pre-release version of hydrated_bloc. Am I missing something?

--- 

Anyways, if anyone is wondering I solved it by using `HydratedBlocOverrides.runZoned` in combination with a custom wrapped testBloc like this:

```dart
class MockStorage extends Mock implements Storage {}

late Storage mockStorage;

void useHydratedMockStorage(VoidCallback fn)  {
  TestWidgetsFlutterBinding.ensureInitialized();
  mockStorage = MockStorage();
  when(
    () => mockStorage.write(any(), any<dynamic>()),
  ).thenAnswer((_) async {});
  HydratedBlocOverrides.runZoned(() => fn, storage: mockStorage);
}

@isTest
void hydratedBlocTest<B extends BlocBase<State>, State>(
  String description, {
  FutureOr<void> Function()? setUp,
  required B Function() build,
  State Function()? seed,
  Function(B bloc)? act,
  Duration? wait,
  int skip = 0,
  dynamic Function()? expect,
  Function(B bloc)? verify,
  dynamic Function()? errors,
  FutureOr<void> Function()? tearDown,
}) {
  test(description, () {
    useHydratedMockStorage(
      () async {
        await testBloc<B, State>(
          setUp: setUp,
          build: build,
          seed: seed,
          act: act,
          wait: wait,
          skip: skip,
          expect: expect,
          verify: verify,
          errors: errors,
          tearDown: tearDown,
        );
      },
    );
  });
}
```

**Comment by felangel** [comment]
@sabifa please upgrade to hydrated_bloc 9.0.0-dev.3. A stable 9.0.0 should be released shortly.
